### ISSUE TITLE
scipts/deps.py only works in the root environment. 

### ISSUE THREAD

**Issue by tyler-thetyrant** [bokeh/bokeh#6620]
[api.render](https://github.com/bokeh/bokeh/blob/0.12.7dev11/scripts/deps.py#L17) only allows parsing the recipe in the root environment.

pyyaml won't be able to do the job as the recipe includes jinja2 template.

**Comment by bryevdv** [comment]
ping @msarahan are there any plans to make conda build installable (or usable?) outside the root env anytime in the near future?

**Comment by msarahan** [comment]
Because conda-build imports conda directly, no, it cannot be used outside of the root environment.  In the absence of a better conda API that allows usage of finer-grained conda functionality outside of the root environment, this is the state of things.

As a workaround, you can use conda-build's ``conda render`` command, which will generate YAML, and you can then load that YAML.

**Comment by bryevdv** [comment]
ok `conda render` takes ~35 seconds to run, so as loathe as I am to do this I think I'm fine just going this route:
```
In [2]: def load_setup_py_data():
   ...:     import os
   ...:     os.environ['CONDA_BUILD_STATE'] = 'RENDER'
   ...:     data = {}
   ...:     def _setup(**kw): data.update(kw)
   ...:     import setuptools
   ...:     setuptools.setup = _setup
   ...:     import setup
   ...:     return data
   ...:

In [3]: import jinja2

In [4]: meta = jinja2.Template(open("conda.recipe/meta.yaml").read())

In [5]: out = meta.render(load_setup_py_data=load_setup_py_data)
```
then parsing the `out` text with `yaml` directly. @tyler-thetyrant would you like to work up a PR to do this?

**Comment by bryevdv** [comment]
FYI here is the current output
```
❯ python scripts/deps.py build run test
python setuptools nodejs>=6.6.0 certifi pycurl packaging six python-dateutil numpy yaml pyyaml jinja2 tornado>=4.3 python six>=1.5.2 pyyaml>=3.10 python-dateutil>=2.1 jinja2>=2.7 numpy>=1.7.1 tornado>=4.3 bkcharts>=0.2 futures-compat pygments sphinx<1.6 colorama coverage flake8 lxml networkx mock packaging pandas pytest>=3.0.2 pytest-cov>=1.8.1 pytest-selenium>=1.6.0 pytest-html selenium>=3 pytest-xdist pytest-rerunfailures requests>=1.2.3 beautifulsoup4 scipy pillow boto nodejs>=6.6.0 notebook sympy scikit-learn networkx icalendar flask flexx
```

**Comment by tyler-thetyrant** [comment]
Closing as the issue has been resolved by #6632
